test/e2e/pvc_test.go

```diff
@@ -0,0 +1,255 @@
+package e2e_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	operatorapi "github.com/openshift/api/operator/v1"
+
+	imageregistryv1 "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1"
+	"github.com/openshift/cluster-image-registry-operator/test/framework"
+)
+
+func testDefer(t *testing.T, client *framework.Clientset) {
+	if t.Failed() {
+		scList, err := client.StorageClasses().List(metav1.ListOptions{})
+		if err != nil {
+			t.Logf("unable to dump the storage classes: %s", err)
+		} else {
+			framework.DumpYAML(t, "storageclasses", scList)
+		}
+
+		pvList, err := client.PersistentVolumes().List(metav1.ListOptions{})
+		if err != nil {
+			t.Logf("unable to dump the persistent volumes: %s", err)
+		} else {
+			framework.DumpYAML(t, "persistentvolumes", pvList)
+		}
+
+		pvcList, err := client.PersistentVolumeClaims(imageregistryv1.ImageRegistryOperatorNamespace).List(metav1.ListOptions{})
+		if err != nil {
+			t.Logf("unable to dump the persistent volume claims: %s", err)
+		} else {
+			framework.DumpYAML(t, "persistentvolumeclaims", pvcList)
+		}
+	}
+	framework.MustRemoveImageRegistry(t, client)
+}
+
+func createPV(t *testing.T, storageClass string) error {
+	client := framework.MustNewClientset(t, nil)
+	name := ""
+
+	for i := 0; i < 100; i++ {
```

Comment:
why do we need to create 100?

Reply:
@bparees You do not quite understand. This is the number of retries.

Reply:
i see now, my bad.